bundles/org.openhab.binding.luxtronikheatpump/src/main/java/org/openhab/binding/luxtronikheatpump/internal/ChannelUpdaterJob.java

```diff
@@ -0,0 +1,285 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.luxtronikheatpump.internal;
+
+import java.io.IOException;
+import java.time.DateTimeException;
+import java.time.Instant;
+import java.time.ZoneId;
+
+import javax.measure.Unit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.luxtronikheatpump.internal.enums.HeatpumpChannel;
+import org.openhab.binding.luxtronikheatpump.internal.enums.HeatpumpType;
+import org.openhab.core.items.Item;
+import org.openhab.core.library.items.DateTimeItem;
+import org.openhab.core.library.items.NumberItem;
+import org.openhab.core.library.items.SwitchItem;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.scheduler.SchedulerRunnable;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ChannelUpdaterJob} updates all channel values
+ *
+ * @author Stefan Giehl - Initial contribution
+ */
+@NonNullByDefault
+public class ChannelUpdaterJob implements SchedulerRunnable, Runnable {
+
+    private final Thing thing;
+    private final LuxtronikHeatpumpConfiguration config;
+    private final LuxtronikTranslationProvider translationProvider;
+    private final Logger logger = LoggerFactory.getLogger(ChannelUpdaterJob.class);
+
+    public ChannelUpdaterJob(Thing thing, LuxtronikTranslationProvider translationProvider) {
+        this.thing = thing;
+        this.translationProvider = translationProvider;
+        this.config = thing.getConfiguration().as(LuxtronikHeatpumpConfiguration.class);
+    }
+
+    public Thing getThing() {
+        return thing;
+    }
+
+    @Override
+    public void run() {
+        // connect to heatpump and check if values can be fetched
+        final HeatpumpConnector connector = new HeatpumpConnector(config.ipAddress, config.port);
+        final LuxtronikHeatpumpHandler handler = (LuxtronikHeatpumpHandler) thing.getHandler();
```

Comment:
This could be moved into constructor including the cast and the non null check so you don't need to initialise the members every time and can assume they are non-null.

Reply:
will change it to a constructor parameter